docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+# mbed TLS Hardware Acceleration
+
+This document explains how to use hardware acceleration for various cryptographic primitives with [mbed TLS](https://github.com/ARMmbed/mbedtls) running on an embedded device.
+
+
+## Introduction
+
+### Why should I add hardware acceleration?
+
+Whether the application developer uses mbed TLS as a cryptographic library or uses it as a TLS stack, they will end up with time-consuming cryptographic routines in their application that will heavily impact response time. The platform that reduces this adverse effect will have an edge over platforms that do not (or do so to a lesser extent).
+
+You may want to add hardware acceleration in the following cases:
+
+
+- You can accelerate parts significantly with optimised assembly code.
+
+
+- Your processor has special instructions capable of accelerating cryptographic operations.
+
+
+- Your processor has access to a co-processor with cryptographic acceleration capabilities.
+
+
+- Your platform has a dedicated crypto-module capable of executing cryptographic primitives, and possibly storing keys securely.
+
+The mbed TLS library was written in C and it doesn’t use any optimised assembly code. The only exception is the arbitrary precision multiplication on selected processors. You can find the list of supported platforms in the top comment in [bn_mul.h](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/bn_mul.h).
+
+### What parts can I accelerate?
+
+mbed TLS has separate modules for the different cryptographic primitives. Hardware acceleration interface is available for the following modules and functions:
+
+- Symmetric
+    - AES: `mbedtls_aes_setkey_enc()`, `mbedtls_aes_setkey_dec()`, `mbedtls_aes_encrypt()`, `mbedtls_aes_decrypt()`
+    - ARC4
+    - BLOWFISH
+    - CAMELLIA
+    - DES: `mbedtls_des_setkey()`, `mbedtls_des_crypt_ecb()`, `mbedtls_des3_crypt_ecb()`
+    - XTEA
+    - MD2: `mbedtls_md2_process()`
+    - MD4: `mbedtls_md4_process()`
+    - MD5: `mbedtls_md5_process()`
+    - RIPEMD160: `mbedtls_ripemd160_process()`
+    - SHA1: `mbedtls_sha1_process()`
+    - SHA256: `mbedtls_sha256_process()`
+    - SHA512: `mbedtls_sha512_process()`
+- Asymmetric:
+    - ECP: `mbedtls_internal_ecp_randomize_jac()`, `mbedtls_internal_ecp_add_mixed()`, `mbedtls_internal_ecp_double_jac()`, `mbedtls_internal_ecp_normalize_jac_many()`, `mbedtls_internal_ecp_normalize_jac()`, `mbedtls_internal_ecp_double_add_mxz()`, `mbedtls_internal_ecp_randomize_mxz()`, `mbedtls_internal_ecp_normalize_mxz()`
+
+<span class="notes">**Note:** The ECP module and function replacement is not part of mbed OS yet. It is in an Early Partner Preview stage and it is available for evaluation in a [PR on github](https://github.com/ARMmbed/mbed-os/pull/3294)</span>
```

Comment:
I am happy to change the phrasing, but it is already in a box as it is. It will look like the note on this page:
https://docs.mbed.com/docs/mbed-os-handbook/en/5.2/advanced/tls_porting/